src/metrics/process.rs

```diff
@@ -0,0 +1,210 @@
+                                            for kv in &kv_list.values {
+                                                new_attributes.push(kv.clone());
+                                            }
+                                        }
+                                    }
+                                });
+                                // Add cx.application.name and cx.subsystem.name attributes
+                                new_attributes.push(KeyValue {
+                                    key: "cx.application.name".to_string(),
+                                    value: Some(AnyValue {
+                                        value: Some(any_value::Value::StringValue(app_name.to_string())),
+                                    }),
+                                });
+                                new_attributes.push(KeyValue {
+                                    key: "cx.subsystem.name".to_string(),
+                                    value: Some(AnyValue {
+                                        value: Some(any_value::Value::StringValue(subsystem_name.to_string())),
+                                    }),
+
+                                });
+                                
+                                debug!("Data Point Attributes: {:?}", new_attributes);
+                                data_point.attributes = data_point
+                                    .attributes
+                                    .iter()
+                                    .filter(|label| label.key != "Dimensions")
+                                    .cloned()
+                                    .collect();
+                                data_point.attributes.extend(new_attributes.iter().cloned());
+                                // Extend data_point attributes with the new attributes
+                                //data_point.attributes.extend(new_attributes.iter().cloned());
+                                debug!("Final DataPoint: {:?}", data_point);
+                            });
+                
+                            // Print the metric after all modifications
+                            debug!("Final Metric Details: {:?}", metric);
+                        }
+                    }
+                }
+            }
+        }
+    };
+
+    Ok(decoded_message)
+}
+
+/// coralogix_send - sends metrics to Coralogix
+async fn coralogix_send(config: &Config, message: Vec<u8>) -> Result<(), Error> {
+    let uri = &format!("{}/v1/metrics", config.endpoint);
+
+    debug!("sending to metric data to uri: {:?}", uri);
+    
+    reqwest::Client::new()
+        .post(uri)
+        .header("Content-Type", "application/x-protobuf")
+        .header(
+            "Authorization",
+            &format!("Bearer {}", config.api_key.token()),
+        )
+        .header("cx.application.name", config.app_name.clone())
```

Comment:
This is not a valid header recognised by metrics-gateway. Besides, the application and subsystem names are already added to the metrics, no need to add them here. (I assume you've verified that the applied name is applied correctly)

Reply:
removed